server/src/main/java/org/apache/iotdb/db/mpp/sql/analyze/Analyzer.java

```diff
@@ -530,6 +533,45 @@ public Analysis visitShowDevices(
       return analysis;
     }
 
+    @Override
+    public Analysis visitCountDevices(
+        CountDevicesStatement countDevicesStatement, MPPQueryContext context) {
+      Analysis analysis = new Analysis();
+      analysis.setStatement(countDevicesStatement);
+      SchemaPartition schemaPartition =
+          partitionFetcher.getSchemaPartition(
+              new PathPatternTree(countDevicesStatement.getPartialPath().concatNode("*")));
```

Comment:
We can use constant `ONE_LEVEL_PATH_WILDCARD` to replace `*` here

Reply:
I will fix it